src/CompanyInfoGazette.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\CompanyInfo;
+
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Http;
+
+class CompanyInfoGazette
+{
+    /**
+     * Lookup a company name on the English Gazette service, return processed company info.
+     *
+     * TODO: Should probably do a more specific query for the name, if possible?
+     *
+     * @param string $name Name of company to lookup.
+     *
+     * @return array|null Array of company info, or null if request to service failed.
+     */
+    public static function lookupName(string $name): ?array
+    {
+        // @phpstan-ignore-next-line
+        $response = Http::withBasicAuth(config('company-info.services.gazette.key'), '')
+            ->get(config('company-info.services.gazette.base_url') . "/search/companies?q={$name}");
+
+        return self::processResponse($response);
+    }
+
+    /**
+     * Lookup a company from number on the English Gazette service, return processed company info.
+     *
+     * TODO: Should probably do a more specific query for the number, if possible?
+     *
+     * @param string $number Number of company to lookup.
+     *
+     * @return array|null Array of company info, or null if request to service failed.
+     */
+    public static function lookupNumber(string $number): ?array
+    {
+        // @phpstan-ignore-next-line
+        $response = Http::withBasicAuth(config('company-info.services.gazette.key'), '')
+            ->get(config('company-info.services.gazette.base_url') . "/search/companies?q={$number}");
+
+        return self::processResponse($response);
+    }
+
+    /**
+     * Process the response from the Virk API call.
+     *
+     * @param Response $response Response.
+     *
+     * @return array|null Array of company info, or null if request to service failed.
+     */
+    private static function processResponse(Response $response): ?array
+    {
+        if ($response->failed()) {
+            return null;
+        }
+
+        $companies = [];
+
+        foreach (Arr::wrap($response->json('items')) as $company) {
```

Comment:
Could we clean this up using collections instead? 

```php
$response
  ->collect('items')
  ->filter(fn (array $company) => $company['company_status'] === 'active')
  ->etc()
```

Reply:
Well phpstan is not happy about this - the Arr::wrap() is there to prevent phpstan from complaining about accessing mixed as an array.

So I keep this, but I change the `if` below around to do a short-circuit `continue` instead.